scripts/prepare-release

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/perl
+##
+## This script will help update manifest.yaml and Changelog.md before a release
+## Any merge to master that changes release.yaml is considered as a new release
+##
+## When ready to make a release, please run ./scripts/prepare-release
+## and commit push the final result!
+use File::Basename;
+use Cwd 'abs_path';
+
+# find its way to the root of git's repository
+my $scriptsdirname = dirname(abs_path(__FILE__));
+chdir "$scriptsdirname/..";
+print "✓ Entering directory:".`pwd`;
+
+# ensures that the current branch is ahead of origin/main
+my $diff= `git diff`;
+chop $diff;
+if ($diff =~ /./) {
+	die("ERROR: Please commit all the changes before calling the prepare-release script.");
+} else {
+	print("✓ All changes are comitted.\n");
+}
+system("git fetch origin");
+my $vcount = `git rev-list --left-right --count origin/main...HEAD`;
+$vcount =~ /^([0-9]+)[ \t]*([0-9]+)$/;
+if ($2>0) {
+	die("ERROR: the current HEAD is not ahead of origin/main\n. Please use git merge origin/main.");
+} else {
+	print("✓ Current HEAD is up to date with origin/main.\n");
+}
+
+mkdir ".changes";
+my $currentbranch = `git rev-parse --abbrev-ref HEAD`;
+chop $currentbranch;
+$currentbranch =~ s/[^a-zA-Z._-]+/-/g;
+my $changefile=".changes/$currentbranch.md";
+my $origmanifestfile=".changes/$currentbranch--manifest.yaml";
+my $origchangelogfile=".changes/$currentbranch--Changelog.md";
+
+my $exit_code=system("wget -O $origmanifestfile https://raw.githubusercontent.com/tfhe/spqlios-fft/main/manifest.yaml");
+if ($exit_code!=0 or ! -f $origmanifestfile) {
+	die("ERROR: failed to download manifest.yaml");
+}
+$exit_code=system("wget -O $origchangelogfile https://raw.githubusercontent.com/tfhe/spqlios-fft/main/Changelog.md");
+if ($exit_code!=0 or ! -f $origchangelogfile) {
+	die("ERROR: failed to download Changelog.md");
+}
+
+# read the current version (from origin/main manifest)
+my $vmajor = 0;
+my $vminor = 0;
+my $vpatch = 0;
+my $versionline = `grep '^version: ' $origmanifestfile | cut -d" " -f2`;
+chop $versionline;
+if (not $versionline =~ /^([0-9]+)\.([0-9]+)\.([0-9]+)$/) {
+	die("ERROR: invalid version in manifest file: $versionline\n");
+} else {
+	$vmajor = int($1);
+	$vminor = int($2);
+	$vpatch = int($3);
+}
+print "Version in manifest file: $vmajor.$vminor.$vpatch\n";
+
+if (not -f $changefile) {
+	## create a changes file
+	open F,">$changefile";
+	print F "# Changefile for branch $currentbranch\n\n";
+	print F "## Type of release (major,minor,patch)?\n\n";
+	print F "releasetype: patch\n\n";
+	print F "## What has changed (please edit)?\n\n";
```

Comment:
At some point we could get the PRs that were merged since the last release

Reply:
On the other hand, we will try to release often, so hopefully, there won't be too much unreleased stuff to remember :)